src/util/android/event_loop_signal.hpp

```diff
@@ -134,6 +167,37 @@ class EventLoopSignal : public std::enable_shared_from_this<EventLoopSignal<Call
         // return 1 to continue receiving events
         return 1;
     }
+
+    // Write a byte to a pipe to notify anyone waiting for data on the pipe
+    static void notify_fd(int fd)
```

Comment:
@bdash 
This is used by apple and epoll. It would be need by all POSIX systems i think. a `posix` dir in the `util`? It does worth it i think... at least `EAGAIN` checking has been fixed on apple version long time ago but it just got discovered recently on Android.

Anyway, i will just merge it like this for now. And create another PR for that.